### PR TITLE
feat(storage:s3): add s3 multi-part upload part size config

### PR DESCRIPTION
**Pull request by jeqo**
Adds a configuration for multi-part upload. Ideally this part size should be the same as chunk size, so we gain some performance when reading[3], but this should be considered later once we have more experience with different chunk sizes.

Resolves: #233

References:

[1] Overview: https://docs.aws.amazon.com/AmazonS3/latest/userguide/mpuoverview.html
[2] Multi-part upload limits: https://docs.aws.amazon.com/AmazonS3/latest/userguide/qfacts.html
[3] https://docs.aws.amazon.com/whitepapers/latest/s3-optimizing-performance-best-practices/use-byte-range-fetches.html
> Typical sizes for byte-range requests are 8 MB or 16 MB. If objects are PUT using a multipart upload, it’s a good practice to GET them in the same part sizes (or at least aligned to part boundaries) for best performance. GET requests can directly address individual parts; for example, GET ?partNumber=N.

### CODE REVIEW

File: storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/S3StorageConfig.java [hunk 1]
```diff
@@ -48,6 +48,14 @@ public class S3StorageConfig extends AbstractConfig {
         + "By default, empty value means S3 library will auto-detect. "
         + "Amazon S3 uses virtual hosts by default (true), but other S3-compatible backends may differ (e.g. minio).";
 
+    private static final String S3_UPLOAD_MULTI_PART_SIZE_BYTES_CONFIG = "s3.upload.multi.part.size.bytes";
```
Comment (ivanyu) [inline storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/S3StorageConfig.java:51]: Maybe just
```suggestion
    private static final String S3_UPLOAD_MULTI_PART_SIZE_BYTES_CONFIG = "s3.upload.part.size.bytes";
```
?

Also I think for consistency with `chunk.size` we should drop `bytes`.

**Review by ivanyu — CHANGES_REQUESTED** [review]
LGTM, only one concern about the property name

**Review by ivanyu — APPROVED** [review]

File: storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/S3StorageConfig.java (hunk 1 above)
Comment (AnatolyPopov) [reply storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/S3StorageConfig.java]: Can this lead to a confusion between chunk and part? IMO having multipart in the name is a little easier to understand but I'm fine with both.

Comment (ivanyu) [reply storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/S3StorageConfig.java]: You're right, good point. `s3.miltipart.upload.part.size` maybe then?

**Review by AnatolyPopov — APPROVED** [review]
LGTM

**Pull request merged**